src/application.js

```diff
@@ -96,9 +96,18 @@ export class Application extends NamedDefaultableInMemoryEntity {
 
     get executables() {
         const tree = getAppTree(this.prop("name"));
-        return Object.keys(tree).map((key) => {
+        const executableList = Object.keys(tree).map((key) => {
```

Comment:
No need to add extra variables; we can do it in one go. Instantiating and then filtering as well as calling `toJSON()` will `deepclone` the executable config unnecessarily; I think we could do sth like this to minimize that:
```javascript
return Object.keys(tree)
    .filter(key) => {
        const { supportedApplicationVersions } = tree[key];
        return (
            !supportedApplicationVersions ||
            (supportedApplicationVersions &&
                supportedApplicationVersions.includes(this.prop("version")))
        );
    })
    .map((key) => {
            return new this.constructor.Executable({ ...tree[key], name: key });
    });
```